### PR TITLE
esp32[c3|h2|c6]: Add support to TWAI/CANBus controller

### PR DESCRIPTION
**Pull request by eren-terzioglu**
## Summary

esp32\[c3|h2|c6\]: Add support to TWAI/CANBus controller

## Impact

ESP32-C3, ESP32-H2 and ESP32-C6

## Testing

`esp32c3-generic:twai` `esp32c6-devkitc:twai` configurations and can example
Note: I tested with CAN loopback mode under the Device Drivers->CAN Driver Support and wired CAN RX to CAN TX with a jumper cable (Default pins are pin 2 and pin 3). Also tested with `ESPRESSIF_TWAI_TEST_MODE` without any cable connection.

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
let's ignore lvgl ci break which is fixed here: https://github.com/apache/nuttx-apps/pull/2370

File: arch/risc-v/src/common/espressif/Kconfig [hunk 1]
```diff
@@ -395,6 +416,104 @@ endif # ESPRESSIF_UART1
...
+config TWAI0_TIMING_125KBITS
+	bool "125 KBits"
+
+config TWAI0_TIMING_250KBITS
+	bool "250 KBits"
+
+config TWAI0_TIMING_500KBITS
+	bool "500 KBits"
+
+config TWAI0_TIMING_800KBITS
```
Comment (acassis) [inline arch/risc-v/src/common/espressif/Kconfig:450]: Why isn't 1MBITS supported? The most common bitrates are: 125K, 250K, 500K and 1M

File: arch/risc-v/src/common/espressif/Kconfig [hunk 2]
```diff
@@ -395,6 +416,104 @@ endif # ESPRESSIF_UART1
...
+config TWAI1_TIMING_125KBITS
+	bool "125 KBits"
+
+config TWAI1_TIMING_250KBITS
+	bool "250 KBits"
+
+config TWAI1_TIMING_500KBITS
+	bool "500 KBits"
+
+config TWAI1_TIMING_800KBITS
```
Comment (acassis) [inline arch/risc-v/src/common/espressif/Kconfig:492]: Ditto

**Review by acassis — APPROVED** [review]

**Comment by acassis** [comment]
@xiaoxiang781216 the CI is breaking because the LVGL update issue:

```
Error: /github/workspace/sources/apps/graphics/lvgl/lvgl/demos/stress/lv_demo_stress.c:81:29: error: format '%zu' expects argument of type 'size_t', but argument 6 has type 'long unsigned int' [-Werror=format=]
   81 |                 LV_LOG_USER("mem leak since start: %zu, frag: %3d %%", mem_free_start - mon.free_size, mon.frag_pct);
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                                       |
      |                                                                                       long unsigned int
/github/workspace/sources/apps/graphics/lvgl/lvgl/demos/stress/../../src/misc/lv_log.h:132:97: note: in definition of macro 'LV_LOG_USER'
  132 | #    define LV_LOG_USER(...) _lv_log_add(LV_LOG_LEVEL_USER, LV_LOG_FILE, LV_LOG_LINE, __func__, __VA_ARGS__)
      |                                                                                                 ^~~~~~~~~~~
/github/workspace/sources/apps/graphics/lvgl/lvgl/demos/stress/lv_demo_stress.c:81:54: note: format string is defined here
   81 |                 LV_LOG_USER("mem leak since start: %zu, frag: %3d %%", mem_free_start - mon.free_size, mon.frag_pct);
      |                                                    ~~^
      |                                                      |
      |                                                      unsigned int
      |                                                    %lu
```

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 the CI is breaking because the LVGL update issue:
> 
> ```
> Error: /github/workspace/sources/apps/graphics/lvgl/lvgl/demos/stress/lv_demo_stress.c:81:29: error: format '%zu' expects argument of type 'size_t', but argument 6 has type 'long unsigned int' [-Werror=format=]
>    81 |                 LV_LOG_USER("mem leak since start: %zu, frag: %3d %%", mem_free_start - mon.free_size, mon.frag_pct);
>       |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>       |                                                                                       |
>       |                                                                                       long unsigned int
> /github/workspace/sources/apps/graphics/lvgl/lvgl/demos/stress/../../src/misc/lv_log.h:132:97: note: in definition of macro 'LV_LOG_USER'
>   132 | #    define LV_LOG_USER(...) _lv_log_add(LV_LOG_LEVEL_USER, LV_LOG_FILE, LV_LOG_LINE, __func__, __VA_ARGS__)
>       |                                                                                                 ^~~~~~~~~~~
> /github/workspace/sources/apps/graphics/lvgl/lvgl/demos/stress/lv_demo_stress.c:81:54: note: format string is defined here
>    81 |                 LV_LOG_USER("mem leak since start: %zu, frag: %3d %%", mem_free_start - mon.free_size, mon.frag_pct);
>       |                                                    ~~^
>       |                                                      |
>       |                                                      unsigned int
>       |                                                    %lu
> ```

Fixed by: https://github.com/apache/nuttx-apps/pull/2370

**Pull request merged**